tests/phpunit/tests/admin/wpUpgrader.php

```diff
@@ -0,0 +1,1430 @@
+<?php
+/**
+ * Tests the `WP_Upgrader` class.
+ *
+ * @package WordPress
+ *
+ * @since 6.1.0
+ */
+
+/**
+ * Tests_Admin_WpUpgrader class.
+ *
+ * @group admin
+ * @group upgrade
+ *
+ * @covers WP_Upgrader
+ *
+ * @since 6.1.0
+ */
+class Tests_Admin_WpUpgrader extends WP_UnitTestCase {
+
+	/**
+	 * An instance of the WP_Upgrader class being tested.
+	 *
+	 * @var WP_Upgrader
+	 */
+	private $instance;
+
+	/**
+	 * @var WP_Upgrader_Skin&\PHPUnit\Framework\MockObject\MockObject
+	 */
+	private $upgrader_skin_mock;
+
+	/**
+	 * Filesystem mock.
+	 *
+	 * @var \PHPUnit\Framework\MockObject\MockObject&WP_Filesystem_Base
+	 */
+	private $wp_filesystem_mock;
+
+	/**
+	 * Loads the class to be tested.
+	 */
+	public static function set_up_before_class() {
+		parent::set_up_before_class();
+
+		require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+		require_once ABSPATH . 'wp-admin/includes/class-wp-filesystem-base.php';
+	}
+
+	/**
+	 * Sets up the class instance for each test.
+	 */
+	public function set_up() {
+		parent::set_up();
+
+		$this->upgrader_skin_mock = $this->getMockBuilder( 'WP_Upgrader_Skin' )->getMock();
+
+		$this->instance = new WP_Upgrader( $this->upgrader_skin_mock );
+
+		$this->wp_filesystem_mock = $this->getMockBuilder( 'WP_Filesystem_Base' )->getMock();
+
+		$GLOBALS['wp_filesystem'] = $this->wp_filesystem_mock;
+	}
+
+	/**
+	 * Tears down the class instance after each test.
```

Comment:
```suggestion
	 * Clean up after each test.
```

Reply:
Slight modification on that, but agreed:
```suggestion
	 * Cleans up after each test.
```